stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/input/MessageInputView.kt

```diff
@@ -527,6 +549,27 @@ public class MessageInputView : ConstraintLayout {
         }
     }
 
+    /**
+     * Creates an instance of [SuggestionListControllerListener].
+     *
+     * Used to modify attachment button visibility to discourage mixing
+     * commands and attachments since they are not compatible.
+     */
+    private fun createSuggestionsListControllerListener(): SuggestionListControllerListener =
+        object : SuggestionListControllerListener {
+            override fun onSuggestionListUiVisibilityChanged(isVisible: Boolean) {
+                isSuggestionListUiVisible = isVisible
+                binding.attachmentsButton.isEnabled = !isSuggestionListUiVisible && !inputContainsCommands
+            }
+
+            override fun containsCommands(doesContain: Boolean) {
+                inputContainsCommands = doesContain
+                binding.attachmentsButton.isEnabled = !isSuggestionListUiVisible && !inputContainsCommands
+            }
+
+            override fun containsMentions(doesContain: Boolean) {}
```

Comment:
Not relevant in our case but it might be handy for users so I decided to put it in the `SuggestionListControllerListener` interface.

Reply:
Sounds good!